node/activation_service.go

```diff
@@ -0,0 +1,712 @@
+	syncer = alwaysSyncedSyncer{}
+
+	atxBuilder := activation.NewBuilder(
+		builderConfig,
+		app.localDB,
+		atxService,
+		atxPublisher,
+		app.validator,
+		nipostBuilder,
+		app.clock,
+		syncer,
+		atxBuilderLog,
+		activation.WithContext(ctx),
+		activation.WithPoetConfig(app.Config.POET),
+		// TODO(dshulyak) makes no sense. how we ended using it?
+		activation.WithPoetRetryInterval(app.Config.HARE3.PreroundDelay),
+		activation.WithPostStates(postStates),
+		activation.WithIdentityStates(app.idStates),
+		activation.WithPoets(poetClients...),
+		activation.BuilderAtxVersions(app.Config.AtxVersions),
+	)
+
+	if len(app.signers) > 1 || app.signers[0].Name() != supervisedIDKeyFileName {
+		// in a remote setup we register eagerly so the atxBuilder can warn about missing connections asap.
+		// Any setup with more than one signer is considered a remote setup. If there is only one signer it
+		// is considered a remote setup if the key for the signer has not been sourced from `supervisedIDKeyFileName`.
+		//
+		// In a supervised setup the postSetupManager will register at the atxBuilder when
+		// it finished initializing, to avoid warning about a missing connection when the supervised post
+		// service isn't ready yet.
+		for _, sig := range app.signers {
+			atxBuilder.Register(sig)
+		}
+	}
+	app.postSupervisor = activation.NewPostSupervisor(
+		app.log.Zap(),
+		app.Config.POST,
+		app.Config.SMESHING.ProvingOpts,
+		postSetupMgr,
+		atxBuilder,
+	)
+	if err != nil {
+		return fmt.Errorf("init post service: %w", err)
+	}
+
+	app.mesh = msh
+	app.syncer = newSyncer
+	app.atxBuilder = atxBuilder
+	app.atxHandler = atxHandler
+	app.poetDb = poetDb
+	app.fetcher = fetcher
+	app.beaconProtocol = beaconProtocol
+
+	return nil
+}
+
+func (app *App) startActivationServiceServices(ctx context.Context) error {
+	if app.fetcher != nil {
+		if err := app.fetcher.Start(); err != nil {
+			return fmt.Errorf("start fetcher: %w", err)
+		}
+	}
+	if app.syncer != nil {
+		app.syncer.Start()
+	}
+
+	if app.beaconProtocol != nil {
+		app.beaconProtocol.Start(ctx)
+	}
+
+	if app.blockGen != nil {
+		app.blockGen.Start(ctx)
+	}
+	if app.certifier != nil {
+		app.certifier.Start(ctx)
+	}
+	if app.proposalBuilder != nil {
+		app.eg.Go(func() error {
+			return app.proposalBuilder.Run(ctx)
+		})
+	}
+	if app.remoteProposalBuilder != nil {
+		app.eg.Go(func() error {
+			return app.remoteProposalBuilder.Run(ctx)
+		})
+	}
+
+	if app.Config.SMESHING.CoinbaseAccount != "" {
+		coinbaseAddr, err := types.StringToAddress(app.Config.SMESHING.CoinbaseAccount)
+		if err != nil {
+			return fmt.Errorf(
+				"parse CoinbaseAccount address on start `%s`: %w",
+				app.Config.SMESHING.CoinbaseAccount,
+				err,
+			)
+		}
+		if err := app.atxBuilder.StartSmeshing(coinbaseAddr); err != nil {
+			return fmt.Errorf("start smeshing: %w", err)
+		}
+	}
+
+	if app.ptimesync != nil {
+		app.ptimesync.Start()
+	}
+
+	if app.updater != nil {
+		app.listenToUpdates(ctx)
+	}
```

Comment:
The updater shouldn't be needed AFAIR (it downloads bootstrapped beacon and activeset, which is needed on the node-service).
```suggestion
```